scripts/legacy.js

```diff
@@ -0,0 +1,214 @@
+// Legacy functions
+// To be removed when vue port is done
+
+import { ALERTS, translation, tr } from './i18n.js';
+import {
+    doms,
+    restoreWallet,
+    toggleBottomMenu,
+    guiSetColdStakingAddress,
+} from './global.js';
+import { wallet, getNewAddress } from './wallet.js';
+import { cChainParams, COIN, COIN_DECIMALS } from './chain_params.js';
+import {
+    createAlert,
+    generateMasternodePrivkey,
+    confirmPopup,
+} from './misc.js';
+import { Database } from './database.js';
+import { getNetwork } from './network.js';
+import { ledgerSignTransaction } from './ledger.js';
+
+/**
+ * @deprecated use the new wallet method instead
+ */
+export async function createAndSendTransaction({
+    address,
+    amount,
+    isDelegation = false,
+    useDelegatedInputs = false,
+    delegateChange = false,
+    changeDelegationAddress = null,
+    isProposal = false,
+}) {
+    const tx = wallet.createTransaction(address, amount, {
+        isDelegation,
+        useDelegatedInputs,
+        delegateChange,
+        changeDelegationAddress,
+        isProposal,
+    });
+    if (!wallet.isHardwareWallet()) await wallet.sign(tx);
+    else {
+        await ledgerSignTransaction(wallet, tx);
+    }
+    const res = await getNetwork().sendTransaction(tx.serialize());
+    if (res) {
+        wallet.finalizeTransaction(tx);
+        return { ok: true };
+    }
+    return { ok: false };
+}
```

Comment:
Slight regression from this modification of the `Send` flow: the wallet no longer requests the user to unlock via a shortcut popup (however, Staking actions still do), the normal "Send" flow instead throws an alert.

A (somewhat lazy) solution is to add:
```js
// Ensure the wallet is unlocked
    if (
        wallet.isViewOnly() &&
        !(await restoreWallet(
            `${translation.walletUnlockTx} ${cChainParams.current.TICKER}!`
        ))
    )
        return;
```

To this function, but I'm aware it's re-used elsewhere too, ideally the Vue component for the Send UI should handle this.